src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -510,6 +512,95 @@ protected function initializeBundles()
     }
 
     /**
+     * Returns an array of bundles to register, with dependencies, ordered.
+     *
+     * @uses appendDependenciesRecursively
+     *
+     * @return BundleInterface[] An array of bundle instances.
+     */
+    protected function registeredDependencies()
+    {
+        $bundles = $this->registerBundles();
+        $children = $rootBundles = array();
+        $hasDependencies = false;
+
+        // Build up bundles as a hash with FQN for basis to rebuild again in correct order given dependencies
+        foreach ($bundles as $bundle) {
+            $children[] = $bundleFQN = get_class($bundle);
+            $rootBundles[$bundleFQN] = $bundle;
+
+            if ($bundle instanceof BundleDependenciesInterface) {
+                $hasDependencies = true;
+            }
+        }
+
+        if (!$hasDependencies) {
+            return $bundles;
+        }
+
+        // Rebuild bundles order with dependencies recursively
+        $bundles = array();
+        $stack = array(get_class($this) => true);
+        $this->appendDependenciesRecursively(
+            $children,
+            $bundles,
+            $rootBundles,
+            $stack
+       );
+
+        return $bundles;
+    }
+
+    /**
+     * Append dependencies of bundles recursively.
+     *
+     * Accepted arguments are as documented below where `string` implies a string with class FQN(Fully Qualified Name),
+     * this string must be in exactly same format as returned by get_class() and PHP 5.5's CLASS constant.
+     * Example of FQN string: "Symfony\Component\HttpKernel\Bundle\BundleInterface"
+     *
+     * @param string[]                $children    Dependencies to apply
+     * @param BundleInterface[string] $bundles     Ordered bundles to append dependencies to
+     * @param BundleInterface[string] $rootBundles Loaded Bundles from registerRootBundles()
+     * @param bool[string]            $stack       For recursion protection, and for debug use on exceptions
+     *
+     * @throws \Symfony\Component\HttpKernel\Exception\DependencyMismatchException On missing dependencies
+     */
+    protected function appendDependenciesRecursively(array $children, array &$bundles, array $rootBundles, array $stack)
+    {
+        while (!empty($children)) {
+            $dependencyFQN = array_shift($children);
+            if (isset($bundles[$dependencyFQN])) {
+                continue;
+            } elseif (isset($stack[$dependencyFQN])) {
+                throw new DependencyMismatchException("Recursive dependency for '{$dependencyFQN}'", array_keys($stack));
+            }
+
+            // If already loaded root bundle, use that to not re instantiate
+            if (isset($rootBundles[$dependencyFQN])) {
+                $dependency = $rootBundles[$dependencyFQN];
+            } else {
+                if (!class_exists($dependencyFQN)) {
+                    throw new DependencyMismatchException("Could not find '{$dependencyFQN}'", array_keys($stack));
```

Comment:
I think that this behavior, as explained in the class, is encouraged by using the `class_exists` check manually in the bundle dependencies registration :confused: